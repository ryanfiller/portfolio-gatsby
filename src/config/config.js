export const pages = [
	{
		name: 'portfolio',
	},
	{
		name: 'about',
	},
	{
		name: 'blog',
	},
	// {
	// 	name: 'r&d',
	// 	url: 'workshop'
	// },
	{
		name: 'contact',
		url: '#contact'
	},
]

export const socialMedia = [
	{
		url: 'http://codepen.io/ryanfiller89/',
		name: 'Codepen',
		color: '#fcd000'
	},
	{
		url: 'https://github.com/ryanfiller',
		name: 'Github',
		// color: '#6cc644'
		color: '#24292e'
	},
	{
		url: 'https://www.linkedin.com/in/ryanfiller',
		name: 'LinkedIn',
		color: '#0077b5'
	},
	// {
	// 	url: 'https://www.facebook.com/ryan.filler',
	// 	name: 'Facebook',
	// 	color: '#3b5998'
	// },
	{
		url: 'https://twitter.com/ryanfiller_',
		name: 'Twitter',
		color: '#1da1f2'
	},
	{
		url: 'https://www.instagram.com/ryanfiller_/',
		name: 'Instagram',
		color: '#bc2a8d'
	},
	{
		url: 'https://dribbble.com/ryanoh',
		name: 'Dribbble',
		color: '#ea4c89'
	},
]