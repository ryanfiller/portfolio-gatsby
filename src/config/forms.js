export const ContactForm = {
    name: 'contact-form',
    fields: {
        name : {
            type: 'text',
            element: 'input',
            required: true,
        },
        email: {
            type: 'email',
            element: 'input',
            required: true,
        },
        message: {
            type: 'text',
            element: 'textarea',
            required: true,
            placeholder: 'Message...',
            extraClassName: 'tall'
        }
    },
    buttons: {
        submit: {
            text: 'Send',
        },
        reload: {
            text: 'Send Another?',
        }
    }
}