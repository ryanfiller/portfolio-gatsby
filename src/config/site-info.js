// https://github.com/LeKoArts/gatsby-starter-portfolio-emilia/blob/master/config/SiteConfig.js

module.exports = {
  
    siteTitle: 'ryanfiller.com', // Navigation and Site Title
    siteUrl: 'https://ryanfiller.com', // Domain of your site. No trailing slash!
    siteLanguage: 'en', // Language Tag on <html> element
    siteLogo: '', // Used for SEO and manifest
    siteDescription: '',
  
    siteFBAppID: '', // Facebook App ID
    userTwitter: '', // Twitter Username
    ogSiteName: '', // Facebook Site Name
    googleAnalyticsID: '',
  
    // Your information
    avatar: '/logos/social.png',
    name: 'Ryan Filler',
    location: 'Memphis',
    socialMedia: [
      {
        url: 'http://codepen.io/ryanfiller89/',
        name: 'Codepen',
        color: '#fcd000'
      },
      {
        url: 'https://github.com/ryanfiller',
        name: 'Github',
        color: '#6cc644'
      },
      {
        url: 'https://www.linkedin.com/in/ryanfiller',
        name: 'LinkedIn',
        color: '#0077b5'
      },
      {
        url: 'https://www.facebook.com/ryan.filler',
        name: 'Facebook',
        color: '#3b5998'
      },
      {
        url: 'https://twitter.com/ryanfiller_',
        name: 'Twitter',
        color: '#1da1f2'
      },
      {
        url: 'https://www.instagram.com/ryanfiller_/',
        name: 'Instagram',
        color: '#bc2a8d'
      },
      {
        url: 'https://dribbble.com/ryanoh',
        name: 'Dribbble',
        color: '#ea4c89'
      },
    ],
  };