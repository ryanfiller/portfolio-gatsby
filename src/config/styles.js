import { injectGlobal } from 'styled-components';

injectGlobal`

  @import url("https://fonts.googleapis.com/css?family=Raleway:300,300i,400,400i,700,700i");
  @import url("https://fonts.googleapis.com/css?family=Dosis:300,400,500,700");


  body {
    margin: 0;
    font-size: 12px;
    font-family: 'Raleway', sans-serif;
    font-weight: 300;

    svg, a {
      transition: .2s;
    }
  }
`;

module.exports = {

  transition: '.2s',

  padding: '2rem',

  colors: {
    white: '#fcfaf6',
    black: '#1e2223',
    blue: '#3f7cac',
    orange: '#f89c7d'
  },

  fonts: {
    sansSerif: {
      fontFamily: '"Raleway", sans-serif',
      fontWeight: '300'
    },
    condensed: {
      fontFamily: '"Dosis", sans-serif',
      fontWeight: '400'
    }
  },

  breaks: {
    large: '1350px',
    tablet: '1024px',
    phone: '500px'
  },

  container: {
    paddingLeft: '2rem',
    paddingRight: '2rem',
  },

  readable: {
    paddingLeft: '2rem',
    paddingRight: '2rem',
    maxWidth: '80rem',
    marginLeft: 'auto',
    marginRight:' auto'
  },
};

// color-blue: #3f7cac;
// color-orange: #f89c7d;
// color-white: #fcfaf6;
// color-black: #1e2223;
// color-gray: lighten($color-black, 10%);
// color-light-gray: lighten($color-gray, 25%);
// color-dark-gray: darken($color-gray, 5%);

// @mixin font-size($desktop-size: 1.5em, $tablet-size: 1.25em, $phone-size: 1em) {

//   font-size: $desktop-size;
//   line-height: 1.25em;

//   @include tablet-break {
//     font-size: $tablet-size;
//   }

//   @include phone-break {
//     font-size: $phone-size;
//   }
// }