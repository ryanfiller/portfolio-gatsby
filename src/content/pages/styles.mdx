---
title: 'styles'
---

# Heading level 1
## Heading level 2
### Heading level 3
#### Heading level 4
##### Heading level 5
###### Heading level 6

Nulla vitae elit libero, a pharetra augue. Maecenas sed diam eget risus varius blandit sit amet non magna. Maecenas sed diam eget risus varius blandit sit amet non magna. Aenean lacinia bibendum nulla sed consectetur.

Cras mattis consectetur purus sit amet fermentum. Vestibulum id ligula porta felis euismod semper. Maecenas sed diam eget risus varius blandit sit amet non magna. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus. Cras mattis consectetur purus sit amet fermentum. Praesent commodo cursus magna, vel scelerisque nisl consectetur et.

*Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.*
_Sed posuere consectetur est at lobortis._
_*Nulla vitae elit libero, a pharetra augue.*_

> Donec ullamcorper nulla non metus auctor fringilla. Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum.

- Donec sed odio dui.
- Morbi leo risus, porta ac consectetur ac, vestibulum at eros.
 -Lorem ipsum dolor sit amet, consectetur adipiscing elit.

1 Donec sed odio dui.
2 Morbi leo risus, porta ac consectetur ac, vestibulum at eros.
3 Lorem ipsum dolor sit amet, consectetur adipiscing elit.