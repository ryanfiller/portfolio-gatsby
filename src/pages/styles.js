import React from "react";
import PropTypes from 'prop-types';

import { graphql } from 'gatsby';

import MarkdownBlock from '../components/markdown-block';

export const query = graphql`
	query StylesPage {
		mdx(frontmatter: { title: { eq: "styles" } } ) {
			frontmatter {
				title
			}
			body
		}
	}
`

const Styles = (props) => {
	
	const post = props.data.mdx.body

	return (
		<MarkdownBlock post={ post } />
	)
}

Styles.propTypes = {
    data: PropTypes.object.isRequired,
};

export default Styles;