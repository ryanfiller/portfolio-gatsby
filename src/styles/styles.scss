@import
    'vendor/module',
    'utility/module',
    'config/module',
    'components/module';