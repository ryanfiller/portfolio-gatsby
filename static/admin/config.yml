backend:
  name: github
  repo: ryanfiller/portfolio-gatsby

publish_mode: editorial_workflow

media_folder: static/images/uploads
public_folder: /images/uploads

collections:
  - name: pages
    label: Pages
    folder: src/content/pages
    create: false
    fields:
      - { name: date, label: Date, widget: date }
      - { name: title, label: Title }

  - name: blog
    label: Blog
    folder: src/content/blog
    create: true
    fields:
      - { name: published,label: Published, widget: boolean, default: false}
      - { name: template, widget: hidden, default: blog-item}

      - { name: title, label: Title, widget: string }
      - { name: excerpt, label: excerpt, widget: text }
      - { name: banner, label: Banner, widget: image }
      - { name: path, label: Url, widget: string }
      - { name: date, label: Date, widget: date }

      - { name: category, label: Category, widget: string }
      - { name: tags, label: Tags, widget: list}

      - { name: body, label: Body, widget: markdown }

  - name: portfolio
    label: Portfolio
    folder: src/content/portfolio
    create: true
    fields:
      - { name: published,label: Published, widget: boolean, default: false}
      - { name: template, widget: hidden, default: portfolio-item}

      - { name: title, label: Title, widget: string }
      - { name: path, label: Url, widget: string }
      - { name: date, label: Date, widget: date }

      - { name: category, label: Category, widget: string }
      - { name: tags, label: Tags, widget: list}

      - { name: client, label: Client, widget: string }
      - { name: clienturl, label: Client Url, widget: string }
      - { name: color, label: Brand Color, widget: string }

      - { name: logocolor, label: White Logo, widget: image }
      - { name: logowhite, label: Color Logo, widget: image }

      - label: "Images"
        name: "images"
        widget: "list"
        fields:
            - {label: Image, name: image, widget: image}

      - { name: body, label: Body, widget: markdown }