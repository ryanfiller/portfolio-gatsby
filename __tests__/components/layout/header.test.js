import Header from '../../../src/components/layout/header'
import Logo from '../../../src/components/logo'
import Navigation from '../../../src/components/layout/navigation'
import Navicon from '../../../src/components/layout/navicon'
import { navBreak } from '../../../src/config/styles'

expect.extend(toHaveNoViolations)

const props = {
  
}

const makeComponent = (props) => (
  mount(
    <ContextMock>
      <Header />
    </ContextMock>
  )
);

describe('<Header />', () => {
  it('should match snapshot', () => {
    const component = makeComponent(props);
    expect(component).toMatchSnapshot();
  })

  describe('responsive layout', () => {

    it('should render the logo and nav when large', () => {
      window.testMediaQueryValues = {width: navBreak + 100};
      const component = makeComponent(props);
      expect(component.find(Logo).length).toBe(1);
      expect(component.find(Navigation).length).toBe(1);
      expect(component.find(Navicon).length).toBe(0);
    })

    it('should render the logo and navicon when small', () => {
      window.testMediaQueryValues = {width: navBreak - 100};
      const component = makeComponent(props);
      expect(component.find(Logo).length).toBe(1);
      expect(component.find(Navigation).length).toBe(0);
      expect(component.find(Navicon).length).toBe(1);
    })
  });

  it('should be (mostly) assecibility', async () => {
    const component = makeComponent(props);
    const html = ReactDOMServer.renderToString(component)  
    expect(await axe(html)).toHaveNoViolations()
  })
});
