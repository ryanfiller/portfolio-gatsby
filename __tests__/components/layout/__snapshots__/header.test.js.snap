// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  position: relative;
  overflow: hidden;
  font-size: 1em;
  height: 1em;
  width: auto;
  color: currentColor;
}

.c1 svg {
  display: block;
  height: 1em;
  width: 1.3em;
  pointer-events: none;
  -webkit-transition: 200ms;
  transition: 200ms;
}

.c1 svg path {
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  fill: currentColor;
  -webkit-transition: 200ms;
  transition: 200ms;
  -webkit-transition-timing-function: steps(2,end);
  transition-timing-function: steps(2,end);
}

.c1 svg mask path {
  color: white;
}

.c1 .r {
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.c1 .yan {
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  opacity: 0;
}

.c1 .f {
  -webkit-transform: rotateY(180deg) translateX(85%);
  -ms-transform: rotateY(180deg) translateX(85%);
  transform: rotateY(180deg) translateX(85%);
  -webkit-mask: url(#mask);
  mask: url(#mask);
}

.c1 .iller {
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  opacity: 0;
}

.c1:hover,
.c1:focus {
  color: #ed6554;
}

.c0 {
  background-color: #fff7ff;
  color: #1e053f;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: calc(2rem / 2);
  font-size: 3rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

@media (min-width:480px) {
  .c1 svg {
    width: 8.45em;
  }

  .c1 svg .r {
    -webkit-filter: none;
    filter: none;
    -webkit-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
  }

  .c1 svg .yan {
    -webkit-transform: translateX(10%);
    -ms-transform: translateX(10%);
    transform: translateX(10%);
    opacity: 1;
  }

  .c1 svg .f {
    -webkit-transform: translateX(50%);
    -ms-transform: translateX(50%);
    transform: translateX(50%);
    -webkit-mask: none;
    mask: none;
  }

  .c1 svg .iller {
    -webkit-transform: translateX(60%);
    -ms-transform: translateX(60%);
    transform: translateX(60%);
    opacity: 1;
  }
}

@media (min-width:768px) {
  .c0 {
    font-size: 2rem;
  }
}

<ContextMock>
  <ThemeProvider
    theme={
      Object {
        "active": "#ed6554",
        "dark": "#1e053f",
        "darkNav": false,
        "disabled": "#515664",
        "highlight": "#7c1863",
        "light": "#fff7ff",
        "primary": "#192368",
      }
    }
  >
    <header__StyledHeader>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "header__StyledHeader-sc-1un6j1h-0",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                [Function],
                " display:flex;justify-content:space-between;align-items:center;padding:calc(",
                "2rem",
                " / 2);font-size:3rem;",
                "
            padding-left: 1rem;
            padding-right: 1rem;
        ",
                " ",
                [Function],
              ],
            },
            "displayName": "header__StyledHeader",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "header__StyledHeader-sc-1un6j1h-0",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <Header
          className="c0"
        >
          <header
            className="c0"
            id="header"
          >
            <logo__StyledLogo
              breakpoint="phone"
            >
              <StyledComponent
                breakpoint="phone"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "logo__StyledLogo-sc-6ahk4j-0",
                      "isStatic": false,
                      "lastClassName": "c1",
                      "rules": Array [
                        "display:inline-flex;justify-content:flex-start;position:relative;overflow:hidden;font-size:1em;height:1em;width:auto;color:currentColor;svg{display:block;height:1em;width:1.3em;pointer-events:none;transition:",
                        "200",
                        "ms;path{transform-origin:center center;fill:currentColor;transition:",
                        "200",
                        "ms;transition-timing-function:",
                        "steps(2, end)",
                        ";}mask path{color:white;}}.r{transform:translateX(0);}.yan{transform:translateX(0);opacity:0;}.f{transform:rotateY(180deg) translateX(85%);mask:url(#mask);}.iller{transform:translateX(0);opacity:0;}&:hover,&:focus{color:",
                        [Function],
                        ";}",
                        [Function],
                      ],
                    },
                    "displayName": "logo__StyledLogo",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "logo__StyledLogo-sc-6ahk4j-0",
                    "target": [Function],
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <Logo
                  breakpoint="phone"
                  className="c1"
                >
                  <ForwardRef
                    aria-label="to homepage"
                    className="c1"
                    to="/"
                  >
                    <GatsbyLink
                      aria-label="to homepage"
                      className="c1"
                      innerRef={null}
                      to="/"
                    >
                      <ForwardRef
                        aria-label="to homepage"
                        className="c1"
                        getProps={[Function]}
                        innerRef={[Function]}
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        to="/"
                      >
                        <Location>
                          <LocationProvider
                            history={
                              Object {
                                "_onTransitionComplete": [Function],
                                "listen": [Function],
                                "location": Object {
                                  "assign": [Function],
                                  "hash": "",
                                  "host": "localhost",
                                  "hostname": "localhost",
                                  "href": "http://localhost/",
                                  "key": "initial",
                                  "origin": "http://localhost",
                                  "pathname": "/",
                                  "port": "",
                                  "protocol": "http:",
                                  "reload": [Function],
                                  "replace": [Function],
                                  "search": "",
                                  "state": null,
                                  "toString": [Function],
                                },
                                "navigate": [Function],
                                "transitioning": false,
                              }
                            }
                          >
                            <a
                              aria-current="page"
                              aria-label="to homepage"
                              className="c1"
                              href="/"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                              style={Object {}}
                            >
                              <test-file-stub />
                            </a>
                          </LocationProvider>
                        </Location>
                      </ForwardRef>
                    </GatsbyLink>
                  </ForwardRef>
                </Logo>
              </StyledComponent>
            </logo__StyledLogo>
            <MediaQueryContextConsumer
              query="(min-width: 768px)"
            >
              <MediaQuery
                query="(min-width: 768px)"
                values={null}
              />
            </MediaQueryContextConsumer>
            <MediaQueryContextConsumer
              query="(max-width: 767px)"
            >
              <MediaQuery
                query="(max-width: 767px)"
                values={null}
              />
            </MediaQueryContextConsumer>
          </header>
        </Header>
      </StyledComponent>
    </header__StyledHeader>
  </ThemeProvider>
</ContextMock>
`;
